libnavigation-core/proguard-rules.pro

```diff
@@ -19,3 +19,18 @@
 # If you keep the line number information, uncomment this to
 # hide the original source file name.
 #-renamesourcefileattribute SourceFile
+
+# --- Navigation SDK Telemetry Events ---
+-keep com.mapbox.navigation.core.telemetry.events.** {*;}
+
+# --- OkHttp ---
+-dontwarn okhttp3.**
+-dontwarn okio.**
+# A resource is loaded with a relative path so the package of this class must be preserved.
+-keepnames class okhttp3.internal.publicsuffix.PublicSuffixDatabase
+
+# --- com.mapbox.api.directions.v5.MapboxDirections ---
+-dontwarn com.sun.xml.internal.ws.spi.db.BindingContextFactory
+
+# --- Telemetry ---
+-keep class com.mapbox.android.telemetry.** { *; }
```

Comment:
This should be handled downstream in the Events library, we shouldn't add this rule (it's not harmful though) cc @harvsu @nkukday

Reply:
Yep, I'm totally agree. However, it exists in legacy SDK, we can remove it if anyone from mobile telemery team approve it

Reply:
Confirmed 👀 https://github.com/mapbox/mapbox-events-android/pull/457